### PR TITLE
[llvm][ARM][NFC] Renaming FeaturePrefLoopLogAlign

### PR DESCRIPTION
**Pull request by nasherm**
The feature 'FeaturePrefLoopLogAlign' was misleading as it was used to set the alignment of branch instructions such as functions. Renamed to FeaturePrefBranchInstAlignment.

### CODE REVIEW

File: llvm/lib/Target/ARM/ARMFeatures.td [hunk 1]
```diff
@@ -372,11 +372,11 @@ def FeatureVMLxForwarding : SubtargetFeature<"vmlx-forwarding",
 def FeaturePref32BitThumb : SubtargetFeature<"32bit", "Prefers32BitThumb", "true",
                                              "Prefer 32-bit Thumb instrs">;
 
-def FeaturePrefLoopAlign32 : SubtargetFeature<"loop-align", "PrefLoopLogAlignment","2",
-                                              "Prefer 32-bit alignment for loops">;
+def FeaturePrefBranchInstAlign32 : SubtargetFeature<"loop-align", "PrefBranchInstLogAlignment","2",
+                                              "Prefer 32-bit alignment for branch instructions (loops, functions)">;
```
Comment (davemgreen) [inline llvm/lib/Target/ARM/ARMFeatures.td:376]: branch instructions -> branch targets

**Review by davemgreen — APPROVED** [review]
Thanks for doing this. I think I would go with "BranchAlign" instead of "BranchInstAlign", which implies that the actual br is aligned.

Otherwise LGTM.

File: llvm/lib/Target/ARM/ARMFeatures.td (hunk 1 above)
Comment (nasherm) [reply llvm/lib/Target/ARM/ARMFeatures.td]: Done

**Review by davemgreen — APPROVED** [review]
Thanks

**Review by tommat01 — COMMENTED** [review]
Personally I would just kept it consistent with the subtarget feature spelling, e.g. `FeatureLoopAlign` to match `loop-align`. But I don't feel that strongly. If you're going to stick with the current suggestion then maybe expand "Pref" to "Prefer" since it only saves two characters.

**Comment by nasherm** [comment]
> Personally I would just kept it consistent with the subtarget feature spelling, e.g. FeatureLoopAlign to match loop-align. But I don't feel that strongly. If you're going to stick with the current suggestion then maybe expand "Pref" to "Prefer" since it only saves two characters.

I've renamed `loop-align` to `branch-align` to keep it consistent. Also changed "Pref" to "Prefer"

**Comment by tmatheson-arm** [comment]
> I've renamed `loop-align` to `branch-align` to keep it consistent. Also changed "Pref" to "Prefer"

Unfortunately you can't do that; it will break any existing IR that refers to this subtarget feature as `loop-align`.

**Comment by nasherm** [comment]
> > I've renamed `loop-align` to `branch-align` to keep it consistent.
> 
> Unfortunately you can't do that; it will break any existing IR that refers to this subtarget feature as `loop-align`.

Ah I see, thanks for pointing this out.

**Pull request merged**